system/modules/core/library/Contao/Picture.php

```diff
@@ -258,16 +259,27 @@ protected function getTemplateDataSource($imageSize)
 				// Use pixel density descriptors if the sizes attribute is empty
 				if (empty($imageSize->sizes))
 				{
-					$src .= ' ' . $density . 'x';
+					if ($fileObj->width && $file1x->width)
```

Comment:
In line 254 you define `$file1x = $fileObj`, which means that if `$fileObj` has a width, `$file1x` will have the same width, right? Then isn't this comparison unnecessary, because the widths will always be the same and the descriptor will always be `1x`?